File: frontend/customerapp/lib/models/location.dart
import 'dart:math';

class Location {
  double _latitude;
  double _longitude;

  Location(
    this._latitude,
    this._longitude,
  );

  double get latitude {
    return _latitude;
  }

  double get longitude {
    return _longitude;
  }

  double getDistanceKm(Location pos2) {
    const double R = 6371.0;
    double latRad1 = _latitude * pi / 180.0;
    double latRad2 = pos2.latitude * pi / 180.0;
    double difLat = (_latitude - pos2.latitude) * pi / 180.0;
    double difLon = (_longitude - pos2.longitude) * pi / 180.0;
    double a = sin(difLat / 2) * sin(difLat / 2) +
        cos(latRad1) * cos(latRad2) * sin(difLon / 2) * sin(difLon / 2);
    double c = 2 * atan2(sqrt(a), sqrt(1 - a));

    return R * c;
  }
}

File: server/glovo_webapi/glovo_webapi/Entities/Location.cs
using System;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace glovo_webapi.Entities
{
    [Owned]
    public class Location
    {
        [Required, Range(-90, 90)]
        public double Latitude { get; set; }
        
        [Required, Range(-180, 180)]
        public double Longitude { get; set; }

        public Location(double latitude, double longitude)
        {
            Latitude = latitude;
            Longitude = longitude;
        }
        
        private const double R = 6371;
        public double DistanceTo(Location location)
        {
            double latRad1 = Latitude * Math.PI / 180.0;
            double latRad2 = location.Latitude * Math.PI / 180.0;
            double difLat = (Latitude - location.Latitude) * Math.PI / 180.0;
            double difLon = (Longitude - location.Longitude) * Math.PI / 180.0;
            double a = Math.Sin(difLat / 2) * Math.Sin(difLat / 2) +
                       Math.Cos(latRad1) * Math.Cos(latRad2) * Math.Sin(difLon / 2) * Math.Sin(difLon / 2);
            double c = 2 * Math.Atan2(Math.Sqrt(a), Math.Sqrt(1 - a));

            return R * c;
        }
    }
}